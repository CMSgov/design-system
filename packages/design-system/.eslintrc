// These rules only apply to files relative to this directory, and allow us
// to apply a stricter set of rules to our package's JavaScript versus the
// documentation site's.
{
  "rules": {
    // We want a displayName for the generated React docs
    "react/display-name": "error",
    // Avoid exploits. If you need dangerouslySetInnerHTML, then temporarily
    // disable this rule in the script rather than removing it from here.
    "react/no-danger": "error"
  }
}
