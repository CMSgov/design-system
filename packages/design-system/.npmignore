**/__tests__/
**/__snapshots__/
**/helpers/
**/*.test.js
**/*.test.jsx
yarn.lock
.stylelintrc.js