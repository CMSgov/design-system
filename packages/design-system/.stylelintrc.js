module.exports = {
  extends: '@cmsgov/stylelint-config-design-system',
  rules: {
    /**
     * Class naming pattern for the core design system
     * ~~~~~~~~~~~~~~~~~~~~
     * Names:    ds-
     * Prefixes: l- c- u-
     * Pattern:  [NAMESPACE]-[PREFIX]-[BLOCK]__[ELEMENT]--[MODIFIER]
     * Examples: .ds-c-button--primary, .ds-c-card__title, .ds-u-text-underlined
     */
    'selector-class-pattern': /^(ds-)(l|c|u|)(-[a-z0-9]+)((--?|__)[a-z0-9]+)*$/,
  },
};
