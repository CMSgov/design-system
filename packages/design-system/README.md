This package contains the following design system assets:

- Base styles
- Utility classes
- Sass/CSS and React components
- Responsive flexbox grid framework
- Fonts and images
- Sass mixins and variables

## Installation

```
npm install --save @cmsgov/design-system
```

## Usage

The source files included are written in Sass (`.scss`). You can add your `node_modules` directory to your Sass [`includePaths`](https://github.com/sass/node-sass#includepaths) and import the file like this:

```css
@import '@cmsgov/design-system/dist/scss/index';
```

or import the transpiled CSS:

```css
@import '@cmsgov/design-system/dist/css/index';
```

[Please view the documentation site for additional information.](https://design.cms.gov/)

## Examples

Examples of the design system in use can be found in the [`examples` directory](https://github.com/CMSgov/design-system/tree/master/examples) on GitHub.

## Bugs and contributions

Please [submit an issue on GitHub](https://github.com/CMSgov/design-system) for any bugs or feature requests.

You can also read our [CONTRIBUTING.md](https://github.com/CMSgov/design-system/blob/master/CONTRIBUTING.md) document to learn about setting up a local development environment, contributing to the design system, and our coding guidelines.

## File Structure

The design system follows a variation of [ITCSS](http://thomasbyttebier.be/blog/less-css-mess) (Inverted Triangle architecture for CSS). The goal is to write CSS in [specificity](https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity) order.

<!-- You can regenerate the tree by running tree -d -I "node_modules" -->

```
└── @cmsgov/design-system/dist
    ├── components/
    │   └── index.js        Compiled JS entry point (CommmonJS)
    ├── css/
    │   └── index.css       Compiled CSS entry point
    ├── esnext/
    │   └── index.esm.js    Compiled JS entry point (ES Module)
    ├── fonts/
    ├── images/
    ├── scss/
    │   ├── base/           Base styles, HTML element selectors
    │   ├── components/     Component styles
    │   ├── settings/       Variables, mixins, and functions
    │   ├── utilities/      Utility classes for individual CSS properties
    │   └── index.scss      Precompiled SCSS entry point
    └── types/              Typescript definition files
```
