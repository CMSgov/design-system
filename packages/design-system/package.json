{
  "name": "@cmsgov/design-system",
  "version": "3.1.0",
  "publishConfig": {
    "access": "public",
    "tag": "latest"
  },
  "description": "CMS Design System Sass/CSS and React assets",
  "repository": {
    "type": "git",
    "url": "https://github.com/CMSgov/design-system",
    "directory": "packages/design-system"
  },
  "license": "SEE LICENSE IN LICENSE.md",
  "main": "dist/components/index.js",
  "module": "dist/esnext/index.esm.js",
  "sideEffects": false,
  "types": "dist/types/index.d.ts",
  "files": [
    "dist",
    "src/components/**/*.{jsx,tsx}",
    "!src/**/*{.test,.spec,.d}.{jsx,tsx}"
  ],
  "dependencies": {
    "@popperjs/core": "^2.4.4",
    "@types/react": "^17.0.10",
    "@types/react-dom": "^17.0.10",
    "classnames": "^2.2.5",
    "core-js": "^3.6.5",
    "downshift": "^3.2.10",
    "ev-emitter": "^1.1.1",
    "focus-trap-react": "^6.0.0",
    "lodash": "^4.17.21",
    "prop-types": "^15.7.2",
    "react-aria-modal": "^2.11.1",
    "react-transition-group": "^4.4.2"
  },
  "peerDependencies": {
    "react": ">=16.0.0",
    "react-dom": ">=16.0.0"
  }
}
