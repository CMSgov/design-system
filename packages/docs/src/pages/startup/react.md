---
title: Using React components
weight: 3
---

The examples below assume you've installed the design system using NPM and have already setup your build system.

<h4 class="ds-h4 ds-u-font-size--base">Default imports</h4>

Individual components can be imported from their individual export file.

```jsx
import Button from '@cmsgov/design-system-core/dist/components/Button/Button';
```

<h4 class="ds-h4 ds-u-font-size--base">Named imports</h4>

Components can also be imported using the shorter syntax below.

<div class="ds-c-alert ds-c-alert--warn ds-u-margin-bottom--2 ds-u-font-size--small">
  <div class="ds-c-alert__body">
    <h3 class="ds-c-alert__heading ds-u-font-size--base">Performance note</h3>
    <p class="ds-c-alert__text">
      This approach may result in a much larger file than you intend. Depending on what module bundler you use, all of the design system's React components may be included in the bundled file even if you didn't specifically import them. This can be avoided by enabling features like [tree shaking in Webpack](https://webpack.js.org/guides/tree-shaking/).
    </p>
  </div>
</div>

```jsx
import { Button } from '@cmsgov/design-system-core';
```

[View an example](https://github.com/CMSgov/design-system/blob/master/examples/react-app/src/scripts/index.js)

<h2 id="examples" class="ds-h2 ds-u-color--primary-darker">Examples</h2>

Additional examples of the design system in use can be viewed on GitHub. These projects demonstrate the various ways you can incorporate the design system into your development process and various use cases.

<a href="https://github.com/CMSgov/design-system/tree/master/examples/" class="ds-c-button">Browse example projects</a>

<h2 id="need-help" class="ds-h2 ds-u-color--primary-darker">Need help or ran into an issue?</h2>

If you're having trouble installing or setting up the design system, or if you think you've found a bug, feel free to [open an issue on GitHub](https://github.com/CMSgov/design-system/issues).
