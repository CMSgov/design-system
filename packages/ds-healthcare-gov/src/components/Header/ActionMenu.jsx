/* eslint-disable react/no-multi-comp, jsx-a11y/no-redundant-roles */
import { Button, CloseIconThin } from '@cmsgov/design-system';
import MenuIcon from './MenuIcon';
import PropTypes from 'prop-types';
import React from 'react';
import classnames from 'classnames';
import { sendHeaderEvent } from './analytics';
import { translate } from 'react-i18next';

const menuId = 'hc-c-menu';

/**
 * ActionMenu is displayed at the top-right of the header.
 * There are two variations of this component: One for logged-out
 * users, which displays a "Log in" & "Espanol" link on desktop.
 * And another for logged-in users, which includes the user's first
 * name. The logged-in variation always displays a "Menu" toggle
 * button, and the logged-out variation only displays it on mobile.
 */
const ActionMenu = function (props) {
  return (
    <nav
      id="hc-c-header__actions"
      className="hc-c-header__actions ds-l-col ds-l-col--auto ds-u-margin-left--auto ds-u-font-weight--bold"
      role="navigation"
    >
      {props.loggedIn ? <LoggedInActionMenu {...props} /> : <LoggedOutActionMenu {...props} />}
    </nav>
  );
};

function LoggedInActionMenu(props) {
  return (
    <div>
      {props.firstName && (
        <span className="ds-u-display--none ds-u-sm-display--inline-block">{props.firstName}</span>
      )}
      <MenuButton {...props} />
    </div>
  );
}

function LoggedOutActionMenu(props) {
  if (props.links.length <= 0) return <></>;

  return (
    <div>
      <ul className="hc-c-logged-out-links ds-c-list--bare ds-u-display--none ds-u-sm-display--inline-block">
        {props.links.map((link) => {
          return (
            <li key={link.href} className="hc-c-logged-out-links__li">
              <a
                href={link.href}
                onClick={() => sendHeaderEvent(link.label, link.href)}
                className="hc-c-logged-out-links__link"
              >
                {link.label}
              </a>
            </li>
          );
        })}
      </ul>
      <MenuButton {...props} className="ds-u-display--inline-block ds-u-sm-display--none" />
    </div>
  );
}

function MenuButton({ t, open, ...props }) {
  const className = classnames(
    'hc-c-action-menu-button',
    'ds-u-sm-margin-left--2',
    props.className
  );

  function onClick(event) {
    sendHeaderEvent(open ? 'menu closed' : 'menu opened');
    props.onMenuToggleClick(event);
  }

  return (
    <Button
      aria-controls={menuId}
      aria-expanded={!!open}
      aria-label={open ? t('header.closeMenu') : t('header.openMenu')}
      className={className}
      onClick={onClick}
      size="small"
    >
      {open ? <CloseIconThin /> : <MenuIcon className="ds-u-margin-right--1" />}
      {t('header.menu')}
    </Button>
  );
}

ActionMenu.propTypes = {
  /** Applies the inverse theme styling */
  firstName: PropTypes.string,
  loggedIn: PropTypes.bool,
  onMenuToggleClick: PropTypes.func.isRequired,
  /**
   * Indicates the menu is open, which influences the label
   * and ARIA attribute of the toggle button.
   */
  open: PropTypes.bool,
  /**
   * These are the links that are visible in the upper left when there
   * is no menu button present. Currently, these only show up when the
   * user is logged out
   */
  links: PropTypes.arrayOf(
    PropTypes.shape({
      identifier: PropTypes.string,
      href: PropTypes.string.isRequired,
      label: PropTypes.node.isRequired,
      onClick: PropTypes.func,
    })
  ).isRequired,
};

LoggedInActionMenu.propTypes = {
  firstName: ActionMenu.propTypes.firstName,
  onMenuToggleClick: ActionMenu.propTypes.onMenuToggleClick,
  open: ActionMenu.propTypes.open,
};

LoggedOutActionMenu.propTypes = {
  onMenuToggleClick: ActionMenu.propTypes.onMenuToggleClick,
  open: ActionMenu.propTypes.open,
  links: PropTypes.arrayOf(
    PropTypes.shape({
      identifier: PropTypes.string,
      href: PropTypes.string.isRequired,
      label: PropTypes.node.isRequired,
      onClick: PropTypes.func,
    })
  ).isRequired,
};

MenuButton.propTypes = {
  onMenuToggleClick: ActionMenu.propTypes.onMenuToggleClick,
  open: ActionMenu.propTypes.open,
  className: PropTypes.string,
};

export default translate()(ActionMenu);
