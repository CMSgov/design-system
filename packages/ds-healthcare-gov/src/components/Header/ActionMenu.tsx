import React from 'react';
import { Button, CloseIconThin, MenuIcon, TFunction } from '@cmsgov/design-system';
import { SyntheticEvent } from 'react';
import classnames from 'classnames';
import { sendHeaderEvent } from './analytics';
import { Link } from './Header';

const menuId = 'hc-c-menu';

export interface ActionMenuProps {
  t: TFunction;
  /** Applies the inverse theme styling */
  firstName?: string;
  loggedIn?: boolean;
  onMenuToggleClick: (event: SyntheticEvent) => any;
  /**
   * Indicates the menu is open, which influences the label
   * and ARIA attribute of the toggle button.
   */
  open?: boolean;
  /**
   * These are the links that are visible in the upper left when there
   * is no menu button present. Currently, these only show up when the
   * user is logged out
   */
  links: Link[];
}

/**
 * ActionMenu is displayed at the top-right of the header.
 * There are two variations of this component: One for logged-out
 * users, which displays a "Log in" & "Espanol" link on desktop.
 * And another for logged-in users, which includes the user's first
 * name. The logged-in variation always displays a "Menu" toggle
 * button, and the logged-out variation only displays it on mobile.
 */
const ActionMenu = function (props: ActionMenuProps) {
  function onClick(event) {
    sendHeaderEvent(props.open ? 'menu closed' : 'menu opened');
    props.onMenuToggleClick(event);
  }

  const menuButton = (
    <Button
      aria-controls={menuId}
      aria-expanded={!!props.open}
      aria-label={props.open ? props.t('header.closeMenu') : props.t('header.openMenu')}
      className={classnames('hc-c-action-menu-button', 'ds-u-sm-margin-left--2', {
        'ds-u-display--inline-block ds-u-sm-display--none': !props.loggedIn,
      })}
      onClick={onClick}
      size="small"
    >
      {props.open ? (
        <CloseIconThin className="ds-u-margin-right--1" />
      ) : (
        <MenuIcon className="ds-u-margin-right--1" />
      )}
      {props.t('header.menu')}
    </Button>
  );

  let content;
  if (props.loggedIn) {
    content = (
      <>
        {props.firstName && (
          <span className="ds-u-display--none ds-u-sm-display--inline-block">
            {props.firstName}
          </span>
        )}
        {menuButton}
      </>
    );
  } else if (props.links.length) {
    content = (
      <>
        <ul
          role="list"
          className="hc-c-logged-out-links ds-c-list--bare ds-u-display--none ds-u-sm-display--flex"
        >
          {props.links.map((link) => {
            return (
              <li key={link.href} className="hc-c-logged-out-links__li">
                <a
                  href={link.href}
                  aria-label={link.ariaLabel}
                  // TODO: .toString() here pacifies TypeScript, but TypeScript has actually found a
                  // potential bug here where we allow link.label to be a ReactNode, but a ReactNode
                  // can't actually be coerced into a string. We've had to do a lot of extra work in
                  // other cases to find the text content of a ReactNode after rendering, like in
                  // packages/design-system/src/components/Alert/Alert.tsx#L114
                  onClick={() => sendHeaderEvent(link.label.toString(), link.href)}
                  className={classnames('hc-c-logged-out-links__link', link.className)}
                >
                  {link.label}
                </a>
              </li>
            );
          })}
        </ul>
        {menuButton}
      </>
    );
  } else {
    content = null;
  }

  return content;
};

export default ActionMenu;
