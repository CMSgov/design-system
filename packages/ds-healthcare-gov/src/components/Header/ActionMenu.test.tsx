import '@testing-library/jest-dom';
import { UtagContainer } from '@cmsgov/design-system';
import ActionMenu from './ActionMenu';
import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';

const t = (key: string) => key;

function renderComponent(props = {}) {
  return render(<ActionMenu t={t} links={[]} onMenuToggleClick={jest.fn()} {...props} />);
}

describe('ActionMenu', function () {
  const handleMenuToggleClick = jest.fn();

  beforeEach(() => {
    handleMenuToggleClick.mockReset();
  });

  describe('logged-in version', () => {
    function renderLoggedIn(props = {}) {
      return renderComponent({
        loggedIn: true,
        firstName: 'John',
        onMenuToggleClick: handleMenuToggleClick,
        links: [],
        ...props,
      });
    }

    it('renders logged-in version', () => {
      const { asFragment } = renderLoggedIn();
      expect(asFragment()).toMatchSnapshot();
    });

    it('set aria-expanded to true', () => {
      const { asFragment } = renderLoggedIn({ open: true });
      expect(asFragment()).toMatchSnapshot();
    });

    it('calls onMenuToggleClick', () => {
      renderLoggedIn();
      fireEvent.click(screen.getByRole('button'));
      expect(handleMenuToggleClick.mock.calls.length).toBe(1);
    });
  });

  describe('logged-out version', () => {
    function renderLoggedOut(props = {}) {
      return renderComponent({
        loggedIn: false,
        onMenuToggleClick: handleMenuToggleClick,
        links: [
          {
            label: 'label',
            href: 'href',
          },
        ],
        ...props,
      });
    }

    it('renders logged-out version', () => {
      const { asFragment } = renderLoggedOut();
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('analytics', () => {
    const mock = jest.fn();

    beforeEach(() => {
      (window as any as UtagContainer).utag = { link: mock };
      jest.clearAllMocks();
    });

    function renderForAnalytics(props = {}) {
      return renderComponent({
        onMenuToggleClick: handleMenuToggleClick,
        ...props,
      });
    }

    it('sends analytics event when logged-out action menu link clicked', () => {
      renderForAnalytics({
        links: [
          {
            label: 'ZOMBO',
            href: 'https://www.zombo.com',
          },
        ],
      });
      fireEvent.click(screen.getByText('ZOMBO'));
      expect(mock).toHaveBeenCalled();
    });

    it('sends analytics event when logged-out menu opened', () => {
      renderForAnalytics({
        links: [{ label: 'label', href: 'href' }],
      });
      fireEvent.click(screen.getByRole('button'));
      expect(mock).toHaveBeenCalled();
      expect(window['utag'].link.mock.calls[0][0]).toMatchSnapshot();
      expect(handleMenuToggleClick).toHaveBeenCalled();
    });

    it('sends analytics event when logged-out menu closed', () => {
      renderForAnalytics({
        open: true,
        links: [{ label: 'label', href: 'href' }],
      });
      fireEvent.click(screen.getByRole('button'));
      expect(mock).toHaveBeenCalled();
      expect(window['utag'].link.mock.calls[0][0]).toMatchSnapshot();
      expect(handleMenuToggleClick).toHaveBeenCalled();
    });

    it('sends analytics event when logged-in menu opened', () => {
      renderForAnalytics({
        loggedIn: true,
      });
      fireEvent.click(screen.getByRole('button'));
      expect(mock).toHaveBeenCalled();
      expect(window['utag'].link.mock.calls[0][0]).toMatchSnapshot();
      expect(handleMenuToggleClick).toHaveBeenCalled();
    });

    it('sends analytics event when logged-in menu closed', () => {
      renderForAnalytics({
        open: true,
        loggedIn: true,
      });
      fireEvent.click(screen.getByRole('button'));
      expect(mock).toHaveBeenCalled();
      expect(window['utag'].link.mock.calls[0][0]).toMatchSnapshot();
      expect(handleMenuToggleClick).toHaveBeenCalled();
    });
  });
});
