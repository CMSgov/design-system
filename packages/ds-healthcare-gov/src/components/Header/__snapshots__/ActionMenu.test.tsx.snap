// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionMenu analytics sends analytics event when logged-in menu closed 1`] = `
Object {
  "category": "consistent header",
  "event_action": "click",
  "event_category": "consistent header",
  "event_label": "menu closed",
  "event_name": "header_click",
  "event_type": "ui interaction",
  "ga_eventAction": "click",
  "ga_eventCategory": "consistent header",
  "ga_eventLabel": "menu closed",
  "ga_eventValue": "",
  "text": "menu closed",
}
`;

exports[`ActionMenu analytics sends analytics event when logged-in menu opened 1`] = `
Object {
  "category": "consistent header",
  "event_action": "click",
  "event_category": "consistent header",
  "event_label": "menu opened",
  "event_name": "header_click",
  "event_type": "ui interaction",
  "ga_eventAction": "click",
  "ga_eventCategory": "consistent header",
  "ga_eventLabel": "menu opened",
  "ga_eventValue": "",
  "text": "menu opened",
}
`;

exports[`ActionMenu analytics sends analytics event when logged-out action menu link clicked 1`] = `
Object {
  "category": "consistent header",
  "event_action": "click",
  "event_category": "consistent header",
  "event_label": "ZOMBO",
  "event_name": "header_click",
  "event_type": "ui interaction",
  "ga_eventAction": "click",
  "ga_eventCategory": "consistent header",
  "ga_eventLabel": "ZOMBO",
  "ga_eventValue": "",
  "link_url": "https://www.zombo.com",
  "text": "ZOMBO",
}
`;

exports[`ActionMenu analytics sends analytics event when logged-out menu closed 1`] = `
Object {
  "category": "consistent header",
  "event_action": "click",
  "event_category": "consistent header",
  "event_label": "menu closed",
  "event_name": "header_click",
  "event_type": "ui interaction",
  "ga_eventAction": "click",
  "ga_eventCategory": "consistent header",
  "ga_eventLabel": "menu closed",
  "ga_eventValue": "",
  "text": "menu closed",
}
`;

exports[`ActionMenu analytics sends analytics event when logged-out menu opened 1`] = `
Object {
  "category": "consistent header",
  "event_action": "click",
  "event_category": "consistent header",
  "event_label": "menu opened",
  "event_name": "header_click",
  "event_type": "ui interaction",
  "ga_eventAction": "click",
  "ga_eventCategory": "consistent header",
  "ga_eventLabel": "menu opened",
  "ga_eventValue": "",
  "text": "menu opened",
}
`;

exports[`ActionMenu logged-in version renders logged-in version 1`] = `
<DocumentFragment>
  <span
    class="ds-u-display--none ds-u-sm-display--inline-block"
  >
    John
  </span>
  <button
    aria-controls="hc-c-menu"
    aria-expanded="false"
    aria-label="header.openMenu"
    class="ds-c-button ds-c-button--small hc-c-action-menu-button ds-u-sm-margin-left--2"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="ds-c-icon ds-u-margin-right--1 ds-c-icon--menu"
      focusable="false"
      id="icon-1"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
        fill="currentColor"
      />
    </svg>
    header.menu
  </button>
</DocumentFragment>
`;

exports[`ActionMenu logged-in version set aria-expanded to true 1`] = `
<DocumentFragment>
  <span
    class="ds-u-display--none ds-u-sm-display--inline-block"
  >
    John
  </span>
  <button
    aria-controls="hc-c-menu"
    aria-expanded="true"
    aria-label="header.closeMenu"
    class="ds-c-button ds-c-button--small hc-c-action-menu-button ds-u-sm-margin-left--2"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="ds-c-icon ds-c-icon--close ds-c-icon--close-thin ds-u-margin-right--1"
      focusable="false"
      id="icon-2"
      viewBox="-2 -2 18 18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 13.0332964L13.0332964 0M13.0332964 13.0332964L0 0"
        fill="none"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-width="2"
      />
    </svg>
    header.menu
  </button>
</DocumentFragment>
`;

exports[`ActionMenu logged-out version renders logged-out version 1`] = `
<DocumentFragment>
  <ul
    class="hc-c-logged-out-links ds-c-list--bare ds-u-display--none ds-u-sm-display--flex"
  >
    <li
      class="hc-c-logged-out-links__li"
    >
      <a
        class="hc-c-logged-out-links__link"
        href="href"
      >
        label
      </a>
    </li>
  </ul>
  <button
    aria-controls="hc-c-menu"
    aria-expanded="false"
    aria-label="header.openMenu"
    class="ds-c-button ds-c-button--small hc-c-action-menu-button ds-u-sm-margin-left--2 ds-u-display--inline-block ds-u-sm-display--none"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="ds-c-icon ds-u-margin-right--1 ds-c-icon--menu"
      focusable="false"
      id="icon-4"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
        fill="currentColor"
      />
    </svg>
    header.menu
  </button>
</DocumentFragment>
`;
